### PR TITLE
[SCB-727] Local generated schema and service center schema are different then print the diff content to the user

### PR DESCRIPTION
**Pull request by maheshrajus**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by liubao68** [comment]
Tips: when adding new commits to existing PR, please use commit message start with issue id, e.g.

[SCB-727] add unit test case

So that we can create a rebase and merge and we also known these commits are related.

File: service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java [hunk 1]
```diff
@@ -498,12 +508,36 @@ public void testLocalSchemaAndServiceCenterSchemaDiff(@Mocked ServiceRegistryCli
...
       }
     };
 
-    microservice.addSchema("s1", "abdc");
+    microservice.addSchema("s1", "abcdefghijklmnopqrstuvwxyz");
     microservice.setEnvironment("prod");
     MicroserviceRegisterTask registerTask = new MicroserviceRegisterTask(eventBus, srClient, microservice);
-    registerTask.run();
+
+    Logger logger = Logger.getLogger(MicroserviceRegisterTask.class);
```
Comment (wujimin) [inline service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java]: use org.apache.servicecomb.foundation.test.scaffolding.log.LogCollector
after finished, do not forget invoke logCollector.teardown()

File: service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java [hunk 2]
```diff
@@ -498,12 +508,36 @@ public void testLocalSchemaAndServiceCenterSchemaDiff(@Mocked ServiceRegistryCli
         result = otherMicroservice;
         srClient.getSchemas(anyString);
         result = onlineSchemasHolder;
+        srClient.getSchema(anyString, anyString);
+        result = "abcdefghijklmn";
```
Comment (wujimin) [inline service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java]: multiple lines schema

File: service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java [hunk 3]
```diff
@@ -498,12 +508,36 @@ public void testLocalSchemaAndServiceCenterSchemaDiff(@Mocked ServiceRegistryCli
...
+    logger.addAppender(appender);
+    Boolean isIlleagalException = false;
+
+    try {
+      registerTask.run();
+    } catch (IllegalStateException exception) {
+      isIlleagalException = true;
+      String logMsg = out.toString();
+      assertNotNull(logMsg);
+      Assert.assertTrue(logMsg.contains("service center schema:"));
```
Comment (wujimin) [inline service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java]: it's better not use contains,  but use equals

File: service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java (hunk 2 above)
Comment (maheshrajus) [reply service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java]: ok..added sample schema file for service center and local

File: service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java (hunk 3 above)
Comment (maheshrajus) [reply service-registry/src/test/java/org/apache/servicecomb/serviceregistry/task/TestMicroserviceRegisterTask.java]: OK

**Pull request merged**